test/archethic_web/graphql_schema_test.exs

```diff
@@ -85,21 +85,38 @@ defmodule ArchethicWeb.GraphQLSchemaTest do
       assert %{"errors" => [%{"message" => "transaction_not_exists"}]} = json_response(conn, 200)
     end
 
-    test "should the transaction with the requested fields", %{conn: conn} do
+    test "should return the transaction with the requested fields", %{conn: conn} do
       addr = <<0::8, 0::8, :crypto.strong_rand_bytes(32)::binary>> |> Base.encode16()
+      prev_public_key = <<0::8, 0::8, :crypto.strong_rand_bytes(32)::binary>>
 
       MockClient
       |> stub(:send_message, fn _, %GetTransaction{}, _ ->
-        {:ok, %Transaction{address: addr, type: :transfer, data: %TransactionData{}}}
+        {:ok,
+         %Transaction{
+           address: addr,
+           previous_public_key: prev_public_key,
+           type: :transfer,
+           data: %TransactionData{}
+         }}
       end)
 
       conn =
         post(conn, "/api", %{
-          "query" => "query { transaction(address: \"#{addr}\") { address } }"
+          "query" => "query { transaction(address: \"#{addr}\") { address, previousAddress } }"
         })
 
-      assert %{"data" => %{"transaction" => %{"address" => address}}} = json_response(conn, 200)
+      assert %{
+               "data" => %{
+                 "transaction" => %{
+                   "address" => address,
+                   "previousAddress" => previous_address
+                 }
+               }
+             } = json_response(conn, 200)
+
       assert addr == Base.decode16!(address, case: :mixed)
+
+      refute is_nil(previous_address)
```

Comment:
Here you can check if the previous address is the expected one using
```elixir
assert Base.decode16!(previous_address) = Crypto.derive_address(prev_public_key)
```

Reply:
thanks